src/Infrastructure.VS/VsVersion/ISetupConfigurationProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.VisualStudio.Setup.Configuration;
+
+namespace SonarLint.VisualStudio.Infrastructure.VS.VsVersion
+{
+    /// <summary>
+    /// Fix for https://github.com/SonarSource/sonarlint-visualstudio/issues/2229 - wrapped instantiation of <see cref="SetupConfiguration"/>
+    /// </summary>
+    internal interface ISetupConfigurationProvider
+    {
+        ISetupConfiguration2 Get();
```

Comment:
It doesn't seem to matter if it's Configuration or Configuration2, the implementing type is the same.

Reply:
I follow a general rule of thumb for bug fixes from my previous employer of "make the minimum change necessary" / "don't change anything you don't need to". Every change has the possibility of introducing some unintended side-effect.
Also, there's no obvious advantage in using `ISetupConfiguration2`, so why change now?

Having said that, I don't think it's worth changing it back.

Reply:
BTW didn't think of this earlier, but we're dealing with COM interop here so there are additional unknowns how it's going to work at runtime. I don't think it will make a difference, but it's an example of how something that looks innocuous could turn out to cause unexpected problems.

Reply:
I'm making a note to better test stuff like that in the future. I knew that this fails in 2015 (hence the try-catch), so I should've tested it on pure 2015 installations.